src/views/Browse/components/Navigation.tsx

```diff
@@ -0,0 +1,51 @@
+import { useMemo } from "react";
+
+import { joinUrl } from "../../../utils";
+
+interface PathPart {
+  name: string;
+  link: string;
+}
+
+export const Navigation: React.FC<{ browseAbsolute: (url: string) => void; path: string }> = ({
+  browseAbsolute,
+  path,
+}) => {
+  const parts: PathPart[] = useMemo(() => {
```

Comment:
We can remove this useMemo as well, If you do want to optimize it you can try 
````
export MyComponent = React.memo(Navigation)`
````

Reply:
Since the whole `Browser` is now wrapped in `React.memo`, any optimization seems superfluous now.